src/newspaperOrderRule.ts

```diff
@@ -158,7 +158,11 @@ abstract class NewspaperHelper {
         const { methodGraph, methodNames } = this;
         try {
             const top = new TopologicalSortUtil(methodGraph);
-            return top.closestList(methodNames);
+            const ordered = top.closestList(methodNames);
+            if (ordered[ordered.length - 1] === '--end-of-dependencies-marker--') {
```

Comment:
@lukaselmer : Please move `'--end-of-dependencies-marker--'` into a reused variable.